### PR TITLE
Add additional metrics

### PR DESCRIPTION
**Pull request by divhead**
- Migrated to batched metrics
- Added metrics for candles loading time
- Added metrics for "trading data" loading time
- Added metrics for markets info loading time
- Added browser info to all metric events
- Added metrics for Long Tasks duration (JS Blocking time)
- Added metrics for Synthetics Page mount time
- Added metrics for Account info initialization time

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208456477170246

### CODE REVIEW

File: src/components/Synthetics/TVChart/TVChart.tsx [hunk 1]
```diff
@@ -15,23 +15,28 @@ import { selectSelectedMarketPriceDecimals } from "context/SyntheticsStateContex
...
 import { TVChartHeader } from "./TVChartHeader";
 
+import { selectSetIsCandlesLoaded } from "context/SyntheticsStateContext/selectors/globalSelectors";
+import { getRequestId, LoadingStartEvent, LoadingSuccessEvent, metrics } from "lib/metrics";
+import { prepareErrorMetricData } from "lib/metrics/errorReporting";
 import "./TVChart.scss";
 
 const DEFAULT_PERIOD = "5m";
+let METRICS_REQUEST_ID: string | undefined = undefined;
+let METRICS_IS_FIRST_TIME_LOAD = true;
```
Comment (gmxer) [inline src/components/Synthetics/TVChart/TVChart.tsx:39]: I think it's weird to name non-constant variables in upper case

File: src/lib/metrics/Metrics.ts [hunk 2]
```diff
@@ -229,12 +164,74 @@ export class Metrics {
...
+      if (this.debug) {
+        // eslint-disable-next-line no-console
+        console.log("Metrics: Slice queue");
+      }
+    }
+
+    let items;
+    if (this.queue.length > MAX_BATCH_LENGTH) {
+      items = this.queue.slice(0, MAX_BATCH_LENGTH);
+      this.queue = this.queue.slice(MAX_BATCH_LENGTH - 1);
```
Comment (gmxer) [inline src/lib/metrics/Metrics.ts:190]: no need to check `this.queue.length > MAX_BATCH_LENGTH`, just always do this logic

**Review by gmxer — CHANGES_REQUESTED** [review]

File: src/lib/metrics/Metrics.ts [hunk 3]
```diff
@@ -229,12 +164,74 @@ export class Metrics {
...
+      if (this.debug) {
+        // eslint-disable-next-line no-console
+        console.log("Metrics: queue is empty");
+      }
+      return;
+    }
+
+    // Avoid infinite queue growth
+    if (this.queue.length > MAX_QUEUE_LENGTH) {
+      this.queue = this.queue.slice(MAX_BATCH_LENGTH - 1);
```
[the remark this answers is not in the document]
Comment (gmxer) [reply src/lib/metrics/Metrics.ts]: not fixed

File: src/lib/metrics/Metrics.ts (hunk 2 above)
Comment (gmxer) [reply src/lib/metrics/Metrics.ts]: not fixed

**Review by gmxer — CHANGES_REQUESTED** [review]
some pending changes left

**Review by gmxer — APPROVED** [review]

**Review by onemicky — APPROVED** [review]

**Pull request merged**